static/js/types/websites.ts

```diff
@@ -12,7 +12,7 @@ export enum WidgetVariant {
 export interface ConfigField {
   name: string
   label: string
-  widget: WidgetVariant
+  widget: string
```

Comment:
Made this change b/c we're now pulling example configs from JSON files, and when typescript analyzes the contents, it just sees a string in this "widget" field. Even though those strings are members of the WidgetVariant enum, it fails. Since our site config schema is defined in the backend and we validate the schema as its created/edited, we can safely remove the type constraint here.

Reply:
sometimes a string literal may need to be casted (i.e. `widgetString as WidgetVariant`). if possible I think it would be nice to keep the typing on the `widget` field more specific than `string` so we can encode the knowledge we have about the possible values it can take in the type system. if it's totally unfeasible to do so then I suppose that's fine, but if we can keep that I think it would be good to do so.

Reply:
I was able to get this to typecheck (leaving `widget` typed as `WidgetVariant`) by just adding a cast to `WebsiteStarterConfig` to the `makeWebsiteStarterConfig` factory function:

```diff
diff --git a/static/js/types/websites.ts b/static/js/types/websites.ts
index 36e129f..e72311f 100644
--- a/static/js/types/websites.ts
+++ b/static/js/types/websites.ts
@@ -12,7 +12,7 @@ export enum WidgetVariant {
 export interface ConfigField {
   name: string
   label: string
-  widget: string
+  widget: WidgetVariant,
   minimal?: boolean
   help?: string
   required?: boolean
diff --git a/static/js/util/factories/websites.ts b/static/js/util/factories/websites.ts
index 15b4cd1..fc5e0d3 100644
--- a/static/js/util/factories/websites.ts
+++ b/static/js/util/factories/websites.ts
@@ -96,7 +96,7 @@ export const makeEditableConfigItem = (name?: string): EditableConfigItem => ({
 })
 
 export const makeWebsiteStarterConfig = (): WebsiteStarterConfig =>
-  cloneDeep(exampleSiteConfig)
+  cloneDeep(exampleSiteConfig as WebsiteStarterConfig)
 
 export const makeWebsiteStarter = (type = "course"): WebsiteStarter => ({
   id:     incr.next().value,
```

that could be an option for keeping the typing, if we decide we want to.

Reply:
Nice! Forgot about `as`. Will fix